src/core/unit-serialize.c

```diff
@@ -172,6 +172,29 @@ static int unit_deserialize_job(Unit *u, FILE *f) {
         return 0;
 }
 
+#define MATCH_DESERIALIZE(key, l, v, parse_func, target)                \
+        ({                                                              \
+                bool _deserialize_matched = streq(l, key);              \
+                if (_deserialize_matched) {                             \
+                        int _deserialize_r = parse_func(v);             \
+                        if (_deserialize_r < 0)                         \
+                                log_unit_debug(u, "Failed to parse \"%s=%s\", ignoring.", l, v); \
+                        else                                            \
+                                target = _deserialize_r;                \
+                };                                                      \
+                _deserialize_matched;                                   \
+        })
+
+#define MATCH_DESERIALIZE_IMMEDIATE(key, l, v, parse_func, target)      \
+        ({                                                              \
+                bool _deserialize_matched = streq(l, key);              \
+                if (_deserialize_matched) {                             \
+                        if (parse_func(v, &target) < 0)                 \
+                                log_unit_debug(u,  "Failed to parse \"%s=%s\", ignoring", l, v); \
```

Comment:
double whitespace...

i'd really pass the return error here btw, even if we don't show it in the text message the journal wil pick it up in structured form, and that's still interesting. if we have an error we should pass it to the log subsystem, regardless if we then reference it in the text string with `%m` or not.